data_prep/roberta_graph_dataset.py

```diff
@@ -1,24 +1,61 @@
+import nltk
+nltk.download('punkt')
+from nltk import word_tokenize
+import torch
+from torchtext.vocab import GloVe
+from transformers import RobertaModel
+
 from data_prep.graph_dataset import GraphDataset
 
 
 class RobertaGraphDataset(GraphDataset):
     """
     Text Dataset used by the Roberta graph model.
     """
-    def __init__(self, *args, **kwargs):
-        # class super init
-        # append features to data object
-        raise NotImplementedError
 
-    def preprocess(self):
+    def __init__(self, corpus, device):
+        super().__init__(corpus, device)
+        self._data.doc_features, self._data.word_features = self._generate_features()
+
+    def _preprocess(self):
         """
         Preprocesses the corpus.
+
+        Returns:
+            tokenized_text (List): List of tokenized documents texts.
+            tokens (List): List of all tokens.
         """
-        # Run Roberta tokenizer on corpus
-        raise NotImplementedError
+        tokenized_text = [word_tokenize(text.lower()) for text in self._raw_texts]
+        tokens = sorted(list(set([token for text in tokenized_text for token in text])))
+        return tokenized_text, tokens
 
-    def generate_features(self):
+    def _generate_features(self):
         """
         Generates node features.
+
+        Returns:
+            features_docs (Tensor): Tensor of document node embeddings.
+            features_words (Tensor): Tensor of token node embeddings.
         """
-        raise NotImplementedError
+        features_docs = []
+        features_words = []
+        doc_embedder = RobertaModel.from_pretrained('roberta-base')
+        token_embedder = GloVe(name='840B', dim=300)
+
+        print('Generating document node features')
+        for text in self._raw_texts:
+            encodings = self._tokenizer.encode(text, truncation=True, padding=False)
+            encodings = torch.tensor(encodings, dtype=torch.long).unsqueeze(0)
```

Comment:
why did you change the squeezing, unsqueezing?

Reply:
Until now it was like this:
```python3
encodings = tokenizer([sentence], truncation=True, padding=False)['input_ids']
encodings = torch.tensor(encodings, dtype=torch.long)
```
Now I use `tokenizer.encode` so I don't need to access `['input_ids']`, which I think is cleaner because `encode` is the function explicitly made for getting `input_ids`. But since encode takes in a single sentence and not a batch, I have to `unsqueeze` outside of the encoder and not "inside" with `[sentence]`.

Reply:
The reason I used `encodings = tokenizer([sentence], truncation=True, padding=False)['input_ids']` is because I tried both (encode and tokenizer()) and it gave me different output. The tokenizer() one gave me the same output as Rahel's model, the encode did not. That's why I opted for tokenizer()